tests/core/pm-module/test_registry_integration.py

```diff
@@ -0,0 +1,219 @@
+import json
+import pytest
+
+from eth_utils import (
+    to_canonical_address,
+)
+
+from web3 import Web3
+from web3.exceptions import (
+    PMError,
+)
+
+try:
+    from ethpm import (
+        ASSETS_DIR,
+    )
+    from ethpm.contract import (
+        LinkableContract,
+    )
+    from ethpm.exceptions import (
+        CannotHandleURI,
+    )
+    import pytest_ethereum as pte
+    from web3.pm import (
+        Registry,
+    )
+except ImportError:
+    ethpm_installed = False
+else:
+    ethpm_installed = True
+
+
+@pytest.fixture
+def w3():
+    w3 = Web3(Web3.EthereumTesterProvider())
+    w3.eth.defaultAccount = w3.eth.accounts[0]
+    w3.eth.defaultContractFactory = LinkableContract
+    try:
+        from web3.pm import PM  # noqa: F811
+    except ModuleNotFoundError as exc:
```

Comment:
Why would this happen?

Reply:
off the top of my head - not sure - this still very much a dirty/wip pr - going through cleanup right now - might be worth waiting for a "ping" to review to avoid some of the weirdness from dev-ing this module on a side branch, but more review is always better

Reply:
mmmm on that note, i just realized that we've dropped py35 in `ethpm` but it's still around in `web3`. so ... it seems as though a merge into master isn't possible until py35 is dropped in `web3`?

Reply:
thinking about the workshop, it adds a little complexity, but is it unreasonable to have the attendees / starter repo work off this `pm-api` branch? (seems unlikely that py35 is gettin dropped in master before devcon)

Reply:
cc @carver web3 master is v5 right?  can we drop 3.5 there?

Reply:
Yup to both! :+1:

Reply:
@njgheorghita I'll leave it to you and @carver to make sure you have a working alpha/beta release of web3-v5 to work with.  That seams reasonable.  If there are any holdups, it should be reasonably easy if not ideal to have them install from source.

Reply:
Yup, I need to do a v4 release, too. I'm flying in about 20 hours, so probably makes sense to wait until I'm there. It seems like a reasonable exception to the weekend release restriction, because I'll be able to dedicate time to checking and fixing anything that pops up. So probably a release of v4 and v5 would go out sometime on Saturday-ish.

Reply:
It would be a v5 alpha, so of course it would have to be `pip install --pre --upgrade web3` to get it. @njgheorghita

Reply:
@carver sounds great! i'm gettin into prague midday on sunday, but will definitely be online saturday